lib/link_state.c

```diff
@@ -306,36 +306,23 @@ int ls_attributes_same(struct ls_attributes *l1, struct ls_attributes *l2)
 	if (CHECK_FLAG(l1->flags, LS_ATTR_USE_BW)
 	    && (l1->extended.used_bw != l2->extended.used_bw))
 		return 0;
-	if (CHECK_FLAG(l1->flags, LS_ATTR_ADJ_SID)) {
-		if ((l1->adj_sid[0].sid != l2->adj_sid[0].sid)
-		    || (l1->adj_sid[0].flags != l2->adj_sid[0].flags)
-		    || (l1->adj_sid[0].weight != l2->adj_sid[0].weight))
+	for (int i = 0; i < LS_ADJ_MAX; i++) {
```

Comment:
Does this mean that we will be storing multiple adjacency SIDs per link? Is this use case really supported in SR drafts?

Reply:
There's no harm though I guess. But I still wonder for which use case scenario will this be used.

Reply:
Well, for a given interface you could have an Adjacency SID + a Backup Adjacency SID for both IPv4 and IPv6 addresses. Thus, if you enable both IPv4 and IPv6 addresses with protection, you ended with 4 Adjacency SID.